src/main/java/com/fryrank/dal/ReviewDALImpl.java

```diff
@@ -36,20 +36,36 @@ public GetAllReviewsOutput getAllReviewsByRestaurantId(@NonNull final String res
         query.addCriteria(equalToRestaurantIdCriteria);
         final List<Review> reviews = mongoTemplate.find(query, Review.class);
 
-        final MatchOperation filterToRestaurantId = match(equalToRestaurantIdCriteria);
-        final GroupOperation averageScore = group(RESTAURANT_ID_KEY).avg("score").as("avgScore");
-        final Aggregation aggregation = newAggregation(filterToRestaurantId, averageScore);
-        final AggregationResults<RestaurantAvgScore> result = mongoTemplate.aggregate(aggregation, "review", RestaurantAvgScore.class);
-        final Optional<RestaurantAvgScore> uniqueResult = Optional.ofNullable(result.getUniqueMappedResult());
+        return new GetAllReviewsOutput(reviews);
+    }
+
+    @Override
+    public GetAggregateReviewInformationOutput getAggregateReviewInformationForRestaurants(@NonNull final List<String> restaurantIds, @NonNull final AggregateReviewFilter aggregateReviewFilter) {
+        final Query query = new Query().addCriteria(Criteria.where(RESTAURANT_ID_KEY).in(restaurantIds));
+
+        Map<String, AggregateReviewInformation> restaurantIdToAggregateReviewInformation = new HashMap<String, AggregateReviewInformation>();
 
-        if (uniqueResult.isPresent()) {
-            final Float rawScore = uniqueResult.get().getAvgScore();
-            final BigDecimal averageScoreBigDecimal = new BigDecimal(rawScore).setScale(1, RoundingMode.DOWN);
+        for(String restaurantId : restaurantIds) {
```

Comment:
you could do restaurantIds.stream().forEach(). It's a functional/immutable way to achieve the same

Reply:
Automatically makes sure everything inside is immutable

Reply:
https://stackoverflow.com/questions/62004870/are-java-8-streams-immutable